### ISSUE TITLE
Publishing a Schedule for Future Releases or a Barebones Project Roadmap

### ISSUE THREAD

**Issue by CharmCityCrab** [fork-maintainers/iceraven-browser#249]
I've noticed that users here and elsewhere seem to occasionally have some anxiety and questions related to when the next update to Iceraven is going to come out.

It occurs to me that perhaps an informal roadmap detailing when major scheduled releases are likely to come out posted here and maintained by one of the project maintainers (Perhaps as a pinned thread that people about to open an issue might see before posting or at least be pointed to by anyone hanging around when they ask.) might be helpful in calming people's nerves.  A link to it could also be posted at the top of the release page itself.

It needn't necessarily be overly complicated.  It could simply be a list of tentative release dates based on when each new stable upstream version is scheduled plus however many days Iceraven developers think it'll usually take after that to have it merged into Iceraven and released with everything working right.

For example, if Firefox 83 is due out on December 1 (I have no idea when Firefox 83 is due out, that date is an entirely made up example), and Iceraven developers think it'll take a week to have an average full version Firefox release merged with Iceraven and released as an Iceraven update, Iceraven 2.0 could be listed as scheduled for December 8th.

Maybe the aim would be to have tentative dates for the next six major planned releases up there at any given time.

A disclaimer could be placed somewhere that says that these are just major releases and some other ones minor uodates may come in between them without notice (i.e. Releases with only new Iceraven-specific features and bug fixes, as well as unscheduled security and bug fixes adapted from unexpected Mozilla releaees), and encouraging users to check back soon.  Since some changes to Iceraven may be major for the browser but only due to the hard work of our developers and thus not on Mozilla's schedule, the Iceraven schedule could in theory include those as well if there is a scheduled end date to a long-term coding project for the browser, with the version numbering for the rest of the schedule after it bumped up accordingly.

The main maintaince tasks that it would create would just be that if a date is going to be missed, people might expect an update and brief explanation like "Postponed from 12/8 to 12/12, it's trickier to integrate the new search engine changes than we thought, but we are hard at work on it." or "Upstream delay. When they are finished, we'll be able to get the next version of Iceraven to you a few days thereafter", and then also indicating putting notations after each version as it's released (and adding minor and unexpected releases retroactively) that say things like "Released 12/8", eventually dropping some old releases off the list and adding new planned future releases on at the end of the list.

It could literally just be a date, the word "Iceraven" and a version number on each line, with the disclaimer at the bottom and message on delays at the top as necessary or something.  It doesn't seem necessary to do a advance release notes for *projected* releases unless someone really wants to. :)

I think people just have these questions in general for any software they run, but they especially have anxiety about small all-volunteer projects sometimes, because such projects sometimes wind up unmaintained without notice.  The devs are doing work on syncing this up to Firefox 83, the devs know that, and other people paying attention know that, but some casual users may see no updates for a month and started to get worried, especially since without a listing on a  Google Play or F-Droid (For now) they are going to be checking back for updates a lot and that could make a gap between updates more noticeable for them than it would be for their usual apps from the Google Play Store.

**Comment by CharmCityCrab** [comment]
@interfect @abhijitvalluri @Dakkaron 

I don't mean to be a pain in the ass, but it's been 44 days since the last release of any kind.  

I count 4 or 5 Firefox for Android updates since then (Not all of them major, of course), the latest being 83.1 nine days ago.

I know a goal was to merge Iceraven with stable upstream releases beginning at v83 and then going forward was discussed at one point.  Is that still the plan?  How close are we?

Is there anything we as users can do to help?  Should we be recruiting more developers so you guys can spread the workload around more?

A lot of users feel invested and want this project to be around for a long time.  So, we're rooting for the devs we do have and trying to help as best we can.

Is there a way we can get more consistent releases by altering the process a little bit?  Once we've got a release based on the stable upstream, that should make regular maintaince and trailing each new Firefox update at a relatively short and predictable interval more doable, right?  

Would basing Iceraven on Fennec (the F-Droid browser, not the abandonware old Firefox codename) help any because of extra codebase similarities?  The latter would put one more layer between the initial updates and Iceraven (i.e. Firefox releases, then Fennec releases, then Iceraven releases rather than Firefox releases and then Iceraven releases.), which is undesirable because we'd theoretically be getting the latest stable security updates and bug fixes on more of a delay, but in a sense the best methodology is whatever the devs feel confident they can commit to getting done regularly.  Right now we haven't had a security update or bug fix in 44 days and we don't know when the next one is coming- so whatever can be done to keep from falling into ruts like this would be good, if the ideal of following every stable Firefox release and update with stable Iceraven releases and updates isn't doable.

**Comment by adityasubathu** [comment]
I also noticed that the last commit in the repo is 17 days old. Has this project been shelved or abandoned?

**Comment by abhijitvalluri** [comment]
Hi all, I have not read all of what was written above, but I do not like the attitude displayed in this thread. Firstly, we are no where near the size of Mozilla, which is a proper organization, with loads of *paid* employees whose day job is to develop Firefox. Sure, it is a non-profit organization, but still the software engineers who work on Firefox are well paid full time employees.

Here, on the other hand, as far as I can tell, this fork was started by @interfect. I then saw this when I was irritated by the latest version of Firefox and how it had terrible issues and regressions with loads of missing features and found an opportunity to update Firefox to have some features I found missing.

I decided to contribute the "fixes" I made so that I can help some others who are also missing these features.

But I never promised any kind of regular updates, neither did @interfect. @Dakkaron definitely showed interest initially but has not done much, but again it is not their fault either as they never promised anything either. None of us are paid to do this work. I have a family and a proper day job and I have been extremely busy with family work during this winter season. Hence I essentially stopped working on this side project.

People should not feel "entitled" for our voluntary work we do in the extra time we get. I spent a lot of time, staying up late in the night doing those commits because I was passionate about fixing Firefox. 

But once I got "Iceweasel or Iceraven" to a point where I found it decent enough for my liking, I stopped working on it as I can not stay late in the night every day to do this when I am not even getting paid for it. I need to take care of my family as well as do my regular day job in the morning.

In fact, I do not even use Iceraven. I use my own "re-forked" version of Iceraven that is rebranded to have "Firefox" logo as I love the Firefox logo. It is for my own personal use and so I do not particularly care in using Firefox logo as it is not intended to be released to anyone else.

That is basically the full story. I have my own life that I need to take care of so I shall not be made to feel "guilty" or obliged to work on anything that I am not personally getting paid for, **and I will especially not be forced into some kind of an agreement or timeline or release schedule that I am not particularly interested in.**

**Comment by CharmCityCrab** [comment]
@abhijitvalluri 

You mentioned in your reply that you had not read the posts in the thread.  If you had, I think that perhaps your reaction would have been less hostile, because they aren't quite what you think they are.

It's true that users want a browser that they can trust to reliably patch security holes and fix bugs.  With all the things a browser is used for in modern society on the modern Internet, I don't think that's unreasonable.

At the same time, though, I tried to make a specific point of being polite, and approaching it from the standpoint of "What can *we* do to help you help us?", essentially.  If there is a problem that we can help you out with, I think you might find some people in the community who would be willing to try.  It's not about blaming anyone, it's about all trying to get to a place where our favorite Android browser is something that remains safe, secure, functional, and, ideally, reliable.

If you guys need more developers, maybe we can help try to recruit them.  If what you want is some sort of monetary compensation for your work, while I am personally living below the poverty line and can't help you there, there are users who might be willing to do that.  I know @interfect mentioned someone sending him some money via PayPal early on in the project's history as a one-time thing.  Maybe other people would be willing to do that if it would help make this worth your while.  I doubt it would pay you what your time is worth at the going rate for a coder of your skill, but if you're still interested in doing this and just need a little something, maybe that's something someone would be willing to toss your way.

Maybe there are other things that we don't know about that would help improve the situation that any one of the three of you can bring up and see if we can help make easier.  We're here to help if we can.

This isn't about calling out you specifically or the trio of maintainers in particular, it's about us loving the concept of Iceraven, and wanting it to be  a project that stays in development rather than potentially being abandoned, which is a reasonable fear at this point, and also being a project that we can feel comfortable using as a default, and recommending to people, that we know is as safe and secure as most browsers are.

The reason your screen name was included with an "at" symbol was because you are one of the three maintainers listed as one of the "fork-maintainers" in the URL.  Obviously, Interfect, who founded the project and is the lead developer is another.  I included @Dakkaron because *I think* he's the third, but I'm not 100% sure on that one.  It's not calling you out, it's just calling an issue to your attention.

I think everyone fully realizes that three people maintaining a browser is not necessarily going to be able to do as much as a browser with six people, especially with the Iceraven trio being three volunteers and the Firefox team being six full-time employees, but, at the same time, the Firefox maintainers have to maintain the core browser essentially in it's entirety, whereas Iceraven can for the most part merge their updates with it's own code and work to fix incompatibilities between the two, which is a hard work, and a lot of work, but not at the same level as maintaining a browser from scratch.

We appreciate all the hard work you and others have done.  I know a lot of us, myself included, have done what we can to promote the browser and give you a user base, and tried to help where we could with bug reports and hooking up code suggested for other Fenix projects with this project so developers have the opportunity to "uplift" certain things from other projects (Either literally other forks of Firefox, or patches designed for Firefox that are good, but rejected by Mozilla) and don't have to do everything Mozilla doesn't do on their own.

But, yes, it is a fair and reasonable expectation that the browser get updated sometimes.  There are bugs that Firefox fixed eons ago that remain broken on Iceraven.  There are almost certainly security vulnerabilities as well.

Just leaving a browser unpatched for months at a time is not an option I would be comfortable with having as a browser I use hours a day every day, or that I'd feel comfortable continuing recommending to others.  If this is abandonware now or soon will be, someone should tell us so we don't compromise our phones and inadvertently keep promoting a project who's developers have moved on.

And, yeah, not blaming you in particular, but it would really suck if Iceraven disappeared or became a "We'll do stuff when we feel like it, maybe, whenever, perhaps every few months" kind of browser, because most of the people here are here because Iceraven is the type of browser we want and we can't get that anywhere else on Android, and we use our phone browsers a lot for important things.

It's reasonable to expect that a web browser will get updates (From someone- not necessarily you, or not necessarily you all the time- but that something be worked out between the developers that keeps someone on it often enough to keep it running smoothly).  If you don't that feel it is, then I think it's your expectations of your userbase's attitudes that are unreasonable IMO.

If there is a problem we can help you with, though, we will try.  Like I said, the point here is to be constructive and try to save the project, not cast blame.  I do not want to go back to Firefox.  I don't really want to switch to Vivaldi for Android as my default as of right now (It's relatively new and still adding features and may one day be a decent choice on Android- it's good on Windows. 
 It's just not great on Android right now, and I am not sure when or if it ever will be).  I'm not sure if the F-Droid Fennec is different enough from Firefox to be worth seriously considering (Though I may start asking around if I *have* to.).

Though I haven't directly coded, I've spent a lot of time on this myself in various ways.

I don't need you telling me I'm the asshole, basically.  I didn't call you the asshole, I asked how we could work with you and the other maintainers to keep this ship from sinking.  If the answer is "Fuck it, I quit.", then I guess what we can do first is see if anyone is interested in replacing you or forking the browser.

A browser is not something that's meant to be developed and then mostly be ignored like a single player video game.  It has to be continuously developed or it puts all of it's users' security at risk and starts to lose compatibility with websites and such.  This browser has been unable to access Twitter links for almost two months.

I appreciate all the hard work you've done and am not suggesting that you need to "work" hours you're uncomfortable with as a volunteer.  I just think the lack of updates and communication are problems, and what I would like to do is work together with the three maintainers and the community to try to find a solution to those issuers (Primarily, the updates.  Communciation is less a problem if there are regular enough updates where we don't have to wonder things like "Wait, is Iceraven still a thing?" or "Can I safely browse the web with this browser anymore?").

If no one is willing to address those type of issues, at some point, I and others will  have to assume the project is abandonware and switch to something that isn't, which isn't a threat to leave- it's a potential sad reality that I really don't want to eventually have to face.  I really genuinely very strongly want this to work out, because no other Android browser is really acceptable to me.  Whatever I might have to switch to would be a compromise I'd dislike a lot but would be forced into because I have an Android phone and I need to have some sort of web browser on it.

So, how can we make this work?

**Comment by CharmCityCrab** [comment]
Can anyone think of a foundation, non-profit, corporation, or other organization that might be interested in helping with maintenance of Iceraven?

**Comment by abhijitvalluri** [comment]
@CharmCityCrab the reason I said I did not read everything is because you often (or rather pretty much always) write things way too elaborately. However, I skimmed through your post quickly as I do not have the luxury of time right now to read them slowly and carefully, and I got the full gist of what you requested. I replied like the way I did above only after I truly felt that your statements were objectionable. 

For your benefit I will more precisely explain what I found objectionable because I do realize that you say all this from a good place and with a lot of passion for the project. However, you have to be mindful of how you say things so that they do not come across poorly to others. Firstly, In the above message you mentioned:

> "It's true that users want a browser that they can trust to reliably patch security holes and fix bugs. With all the things a browser is used for in modern society on the modern Internet, I don't think that's unreasonable."

You keep claiming "users are demanding xyz" or "users are expecting xyz", or "users are requesting xyz", etc. It is super strange the way you say these things, because to the best of my knowledge it is mostly you who keeps saying this. I never heard from any other users of this forked browser, requesting or expecting updates or other such claims. Are you projecting your own ideas onto others as well? Or are you genuinely conducting surveys to gauge user requirements? This is an annoying behavior that I find frustrating sometimes, especially as it sounds demanding and we are doing this in our **very** limited spare time! I mean really limited. I may get at best 1 hour or so in a week to even think about this project and these days that is down to zero.

Moreover, just like how you found my reply "hostile", I found your first post and second post to be very accusatory. Making statements such as:
> I don't mean to be a pain in the ass, but it's been 44 days since the last release of any kind.

That sentence sounds extremely ungrateful to put it mildly.  

To give you some context, had this been the behaviour of my product manager at my office towards his engineers, it would be considered rude and our product manager would be disciplined or at the very least would have been informed to correct his demanding behavior. **And note that that is what would happen at my work for which I am being paid**.

At the end of the day, this was never really claimed to be a well maintained and updated project. And I am not even sure why I am currently wasting my time replying to issues on this project, rather than spending time with my family and sleeping (it is past midnight here!)

Hence, first have some **self-reflection** and realise that your own behavior can come across as demanding/ungrateful  before commenting on others (specifically my response being hostile).

**Comment by abhijitvalluri** [comment]
Finally, I will say one last thing:

There is an "unwritten rule" that is often stated but regularly ignored by users of these kinds of projects, which is projects that are done by a tiny team or group of people on the web for fun, or out of self-interest, and the rule goes like this:

**NEVER ASK FOR AN ETA & NEVER PESTER THE DEVS!** 😄 

And it is very true! A lot of times, I was on your side @CharmCityCrab. I too was a user of Android ROMs and sometimes I too was unable to contain my excitement and inadvertently pestered the devs a bit too much, asking when the next update is due, asking if I can help, and in general being a bit annoying to the devs! It is natural, but if you are in my place right now, you too will realize how frustrating it can be and how frustrating it can feel to be constantly bombarded with emails that can sound demanding. Because that means spending less time with family trying to maintain an open-source project, spending less time relaxing and getting rest. Because I do software engineering during the day. If I do coding projects in the evening as well then I will feel burned out as I already do that for work! I would love to do something different during play time, such as watch TV, play Gran Turismo Sport (PS4 game), or go for a walk etc.

This may sound like a digression, but it is relevant. It is almost impossible to commit to any formal or informal roadmap when it is a passion project and not something you truly wish to spend 4-5 hours at least per day to maintain properly.

So, anyway, get some rest @CharmCityCrab and let me sleep now! lol. And don't worry about these roadmaps etc, if you wish to use this browser, go for it. If you are disappointed that it is not well maintained, I am sorry but I won't exert myself unnecessarily just for you (I am being honest and realistic). And if you really care for this project, perhaps start learning Android programming and get coding yourself! If you have loads of free time then that can be a great project for you to get into and contribute in a proper manner...

Alright, that is all from me for now. Oh man!! I messed up again and I am gonna sleep late due to this... should've read your email in the morning! oh well.

**Comment by CharmCityCrab** [comment]
> You keep claiming "users are demanding xyz" or "users are expecting xyz", or "users are requesting xyz", etc. It is super strange the way you say these things, because to the best of my knowledge it is mostly you who keeps saying this. I never heard from any other users of this forked browser, requesting or expecting updates or other such claims. Are you projecting your own ideas onto others as well? Or are you genuinely conducting surveys to gauge user requirements? 

My initial post at the top of the thread has 13 upvotes and 1 downvote (The downvote is you- and, to be fair, one of the upvotes is me, so that puts it was 12-0 discounting our own votes).  I am not saying that's a scientifically valid sample or something, but that method was what was used when we when voting on browser logos and might be the best we have short of the thread becoming more active or something being devised to poll the entire user base.

Actually, in a way it's worrying that there aren't nearly as many people here chiming in as there used to be.  That in and of itself could be seen as a danger sign that momentum for the project is slowing.  Whether that's because of the relative infrequency of the updates or not is anyone's guess, but that could be part of it.

So, that's part of why I say "we" sometimes.

>This is an annoying behavior that I find frustrating sometimes, especially as it sounds demanding and we are doing this in our **very** limited spare time! I mean really limited. I may get at best 1 hour or so in a week to even think about this project and these days that is down to zero.

Let me try to clarify a little:

Not everything I say is directed at you in particular.  I mean the project in general.  That would be in the most minimalistic sense, the trio of maintainers of which you are one-third, and could also be taken as the three of you plus anyone who is or could be contributing beyond that, etc..  No one is expecting you to do this all yourself and I do realize you are a volunteer who has no formal obligations here and could quit at any time.

If even that places too much of a demand on you, I can understand that, too.  I really do appreciate all the hard work you've done.  That's why part of what I have raised for discussion is what we non-developer users can do.  If this is your limit and the two other people's limit, I am not going to argue with that.  You guys know yourselves and your own lives much better than I do.  I'm just wondering if there is a way for me and anyone else who would like to see more regular (or at least more predictable) releases to get that *without* bugging you guys about it.  For example, if we found you a fourth developer, would that be useful, or would it get in your way?  That kind of thing.

> > I don't mean to be a pain in the ass, but it's been 44 days since the last release of any kind.
> 
> That sentence sounds extremely ungrateful to put it mildly.

I am not always great with phrasing things in a way that people find as polite as I intend them to be.  When I typed that, I thought that by saying "I don't mean to be a pain in the ass", it would convey that I knew I could be perceived as a pain in the ass and that I was somewhat apologetic from the onset about even bringing up the topic and was doing so only reluctantly because I perceived it to be important.  I fear it may have been read instead as me being hostile because I used a phrase that some people use when they're mad that contained a mild obscenity.  My poor communications skills may have caused me to type something to spin it more nicely that actually did the opposite.  If you're offended, I apologize.  I am not trying to offend you or anyone, I am trying to see if there is a game plan that can be devised (By anyone) to make this thing work, is all.  It's not driven by negative feelings towards you or anyone else.

I am concerned about the project- not in accusatory way, in a way that just means in general I wonder if it'll be around in 6 months.  I would like it to be, so I'm trying to see what if anything can be done by anyone.  Maybe the answer is that nothing can be, that we don't have anyone with the right skills and enough time to keep this up anymore.  However, I think it's worth exploring alternatives to see if there is any way around that- and by that I mean anyone who wants to brainstorming about alternatives, I'm not asking you or anyone to rearrange their lives.
 
> To give you some context, had this been the behaviour of my product manager at my office towards his engineers, it would be considered rude and our product manager would be disciplined or at the very least would have been informed to correct his demanding behavior. **And note that that is what would happen at my work for which I am being paid**.

There are a lot of reasons I am not in a position like project manager.  That is probably one of them.  I mean well, but interpersonal dynamics are not my strong suit.

**Comment by CharmCityCrab** [comment]
> Finally, I will say one last thing:
> 
> There is an "unwritten rule" that is often stated but regularly ignored by users of these kinds of projects, which is projects that are done by a tiny team or group of people on the web for fun, or out of self-interest, and the rule goes like this:
> 
> **NEVER ASK FOR AN ETA & NEVER PESTER THE DEVS!** 😄
> 
> And it is very true! 

I suppose I can see why some might feel that is a good rule.  The main concern I'd have about it is just that at a certain point it is ask the question or assume the answer is no and switch to another piece of software without comment.  I thought it was better to bring it up, because perhaps the answer was "Behind the scenes we have divided up the workload, and here is what we expect.".  Perhaps I would switch browsers and find that Iceraven was updated the next day or the next week and since I like this browser the best, I would have been happier staying, but never found out that there was this update coming or that it landed.

I think a lot of people do walk away from software for that reason and would have stayed had they known what was going on, but they never asked, so they did not know.

It also wasn't immediately obvious to me whether a realistic release schedule based on the amount of work to be done and the time each developer was willing to allot to it (Without pressuring anyone to add more than they already planned to do) was something that the folks working on the project didn't put up because they felt philosophically opposed to it, because they had just never thought it over before, or because they thought people wouldn't be interested and wouldn't help with user retention.  It even seemed possible to me that such a roadmap already existed behind the scenes and wasn't being made public because of a perceived lack of interest.  So, I chimed in to say I would be interested in it, in case it was just assumed no one would be.

I think this thread is giving me the gist of where you stand on these subjects.  Maybe one or both of the other two gentleman will chime in at some point as well.

>And if you really care for this project, perhaps start learning Android programming and get coding yourself! If you have loads of free time then that can be a great project for you to get into and contribute in a proper manner...

If I thought I could reasonably develop those skills, I would.  Every attempt I've made to learn a foreign language or programming code has ended in abysmal failure.  I just don't seem to have the aptitude.  The closest I came is that I used to know some very simple early HTML like how to add a link or an image on a Geocities page.

There have been many times where I've thought that if I knew a little programming, I could get so much more of what I want out of my devices and software.  But just like not everyone can be a good basketball player, not everyone can be a good programmer.  Work and time and effort have some impact, but natural skill and talent or lackthereof are also elements.

I did write an initial draft of this project's readme.md, which was heavily altered by interfect before first being posted, but clearly writing at length is something I can do.  If you ever want something like that again, I could help that way.  It's very unlikely I will ever be able to learn to code to the level of where I could help with the actual code itself, though, unless finding good patches written by others for other Fenix-based products and calling them to Iceraven's people's attention counts- I've done that a few times.

**Comment by CharmCityCrab** [comment]
Okay, last thing from me for a while, simple question, but no one should feel the need to answer if doing so will anger him or her-

Is there a plan of some kind for future development, or is this just a thing where if someone finds the time they will do a little work on what at that time they think needs to be done, and if there isn't the time, we'll just float around without updates for a few months or even forever regardless of the security issues, intensifying bugs, or web compatibility issues that may create?

If you choose to answer, please keep in mind I am only asking this to try to get a sense as to whether this is the right browser for me anymore.  I am not saying it to criticize anyone.  I just am trying to figure out if this is simply a lull, a browser going slightly slower than my preferences, or a browser that basically is going to be abandonware aside from the occasional burst of development activity that may or may not come occasionally and relatively randomly.

**Comment by abhijitvalluri** [comment]
> Okay, last thing from me for a while, simple question, but no one should feel the need to answer if doing so will anger him or her-
> 
> Is there a plan of some kind for future development, or is this just a thing where if someone finds the time they will do a little work on what at that time they think needs to be done, and if there isn't the time, we'll just float around without updates for a few months or even forever regardless of the security issues, intensifying bugs, or web compatibility issues that may create?

Unfortunately, I believe that is indeed the situation right now. We never really tried to have a serious plan. I agree that it would be nice to do, but it appears to be the case where either interfect or I will do some updates if we get time. And a few people will contribute PRs if they get time. 

> 
> If you choose to answer, please keep in mind I am only asking this to try to get a sense as to whether this is the right browser for me anymore. I am not saying it to criticize anyone. I just am trying to figure out if this is simply a lull, a browser going slightly slower than my preferences, or a browser that basically is going to be abandonware aside from the occasional burst of development activity that may or may not come occasionally and relatively randomly.

It definitely makes sense to reconsider this as your default browser or not. I am thinking the exact same thing as you are. Because I really do not have much time nowadays, I too am wondering if I should simply go back to official Firefox for convenience/security reasons or if I should try to update this at some regular intervals for my own sake (and of course for the benefit of those who end up using it).

**Comment by abhijitvalluri** [comment]
My main 2 gripes with official FF is the tab tray layout and lack of some addons that I needed. So I did those updates and once I was satisfied, I was not really fully motivated to keep this going. I did try my best to ease the process for anyone who wished to keep it going, by automating a few things (adding checks in PRs, automating the release process via Github actions). 

But it is definitely a challenge to keep this up-to-date with official Firefox even then, as we can get a lot of merge conflicts due to our own changes.

### UPDATE:
So I looked into Firefox nightly, and as of now they have enabled the "Grid view" in the tabs tray that was heavily inspired by my PRs over here! And there are enough add-ons supported officially that I don't need anything else. And they also have added a "Custom addons collection" setting, also likely inspired by my work here! 

So, for me personally, I am now happy with official Firefox Nightly and I will help them to improve it by contributing to the official FF repository. I am also delighted that our work here inspired actual and noticeable improvements in official Firefox as well!

**So, my status update is I plan to contribute bug fixes/improvements to the official https://github.com/mozilla-mobile/fenix project to improve the official Firefox! 😄** I am delighted that my work on this project helped inspire and made Mozilla officially include a version of the Grid view layout of the tabs tray! So, we all should be proud that our small fork helped improve FF and make it better for the millions of users who use the official version! 🥳 🎊 🚀 

I think my time is better spent helping fix the official Firefox, if there are other such usability concerns. So, I shall no longer work on this project (99% sure about that). Others are obviously welcome to continue working on this project and make Iceraven a wonderful browser to use as well, but I am sadly not able to continue. Family comes first obviously.

**Comment by CharmCityCrab** [comment]
I appreciate the candor of the above post and I do think that even if this winds up being the end of Iceraven, it will have a legacy of having influenced Firefox for the better (Incidentally, if anyone can get them to add protocol and "www" as an option in the GUI the way it is in Iceraven, that would be helpful.  I almost feel that it's not there out of spite at this point.).

However, Firefox is still not where I want it to be, and, even if nightly were (Which it isn't), it is hard for me to think a scenario where I would let myself use use an alpha or beta level browser from a big company as my default.  It's one thing if it's a small effort like Iceraven is and where I sort of accept that some extra bugs may get through because the resources aren't there to do a whole bunch of testing on a wide variety of devices, but Mozilla certainly has those resources *and* is deliberately holding back certain features from stable because they don't think their regular users will be able to use them responsibility (Or something like that.).

I actually was very happy to see that Iceraven was planning to base itself on stable but with it's own features and some features from nightly and beta ported in, because it would have given us more than any mobile version of Firefox that's current has, while still having a stable base.  However, it *appears* the project fell apart before we got there.

I'm still kind of waiting for interfect to say something before I give up on Iceraven completely, but I will probably be opening an issue shortly to see if we can find anyone who is willing to pick up the ball on this if the current group of devs no longer have the interest- either by taking over the current project, name, trademark, and all, if they're allowed to, or simply by forking it and developing it onward by a new name in a new repo if necessary.

**Comment by interfect** [comment]
Thanks for all the help @abhijitvalluri! Honestly I think if we can talk Mozilla into taking *some* way to let people add arbitrary extensions, Iceraven won't really need to exist.

As for a release schedule, I would love to target making our own releases at or soon after Mozilla's corresponding ones, as a general plan. But, unlike Mozilla, nobody actually has the job of working on this project, so they'll come out when they come out.

If @abhijitvalluri isn't going to hang around here anymore, I don't necessarily have to actually keep CI working, so I can go back to cutting releases that are basically the same builds that I actually run, instead of trying to polish them up to keep CI happy.

If other people want to volunteer to be a new wave of maintainers, I would be very pleased!

**Comment by Dakkaron** [comment]
I am a bit late to the party, but I'll still chime in.

I just switched jobs and I joined this project while I was using up my left-over vacation days. Now that I started my new job, time has been a very rare commodity, similar to @abhijitvalluri.

Also, I mostly wanted to contribute here, because it would meet my personal needs. I don't have much time to spare to make the world a better place. If I build something for myself and someone else can benefit from that, I am happy about that, but it's not my main goal.

Firefox Nightly includes most of the features I really wanted from IceRaven (namely custom plugin collections and about:config), and it isn't much less stable than IceRaven. So I switched over to that for a while, but now I downgraded back to the much superior Firefox 68. And that is where I get the biggest benefit for the lowest cost.

I actually don't think I should be a maintainer here, since I don't have the time to spare and IceRaven doesn't benefit me enough to actually spend lots of time on it.

When asking for work on a hobbyist open source project consider this: An average senior software developer in Europe earns around €25-40 an hour. Say, you ask for a regular release, say every two weeks. A release involves at least pulling upstream, merging, building and releasing. That's the bare minimum of work. That takes, say 2-3h. So you ask the developers to donate around €1300 to €3120 of work time every year.
This does not include any work needed for stuff like CI, merging pull requests or even developing features on their own.

That is why developers on hobbyist projects usually contribute work for themselves, because spending this much time (and thus indirectly money) is only justified if you really want the result for yourself.

Adding developers wouldn't help much, since more people means more overhead. If, for example, you have 10 developers and you want to maintain the release schedule of one release every two weeks, you need to plan around 2 hours a week for every developer to synchronize their work. So by adding 10 developers, you are effectively wasting 10 hours a week of work time for basically nothing.

Donating a few Euros here or there is nice, but doesn't really finance anything.

There is a reason why there are hardly any successful open source projects that don't have a company behind it, that actually finances the development.

@interfect If you want to clean up, feel free to unmaintainer me. Realistically speaking, I fear I don't have the time to really be helpful here.

**Comment by akliuxingyuan** [comment]
Iceraven is and will be based on Fenix stable release, I will update Iceraven to follow upstream within my ability.
